experimental/relativetime/src/format.rs

```diff
@@ -0,0 +1,94 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::fmt::Write;
+
+use fixed_decimal::FixedDecimal;
+use writeable::Writeable;
+
+use crate::{
+    options::{Numeric, RelativeTimeFormatterOptions},
+    relativetime::RelativeTimeFormatter,
+};
+
+pub mod parts {
+    use writeable::Part;
+
+    /// The [`Part`] used by [`FormattedRelativeTime`](crate::FormattedRelativeTime) to mark the
+    /// part of the string that is without a placeholder.
+    pub const LITERAL: Part = Part {
+        category: "relativetime",
+        value: "literal",
+    };
+}
+
+/// An intermediate structure returned by [`RelativeTimeFormatter`](crate::RelativeTimeFormatter).
+/// This structure can be consumed via [`Writeable`](Writeable) trait to a string or buffer.
+pub struct FormattedRelativeTime<'a> {
+    pub(crate) formatter: &'a RelativeTimeFormatter,
+    pub(crate) options: &'a RelativeTimeFormatterOptions,
+    pub(crate) value: FixedDecimal,
+    pub(crate) is_negative: bool,
+}
+
+impl<'a> Writeable for FormattedRelativeTime<'a> {
+    fn write_to_parts<S: writeable::PartsWrite + ?Sized>(&self, sink: &mut S) -> core::fmt::Result {
+        if self.options.numeric == Numeric::Auto {
+            let relatives = &self.formatter.rt.get().relatives;
+            if let Ok(i8_value) = self.value.to_string().parse::<i8>() {
```

Comment:
Just noticed this: we shouldn't convert to a string and parse here. Better would be something along the lines of

```
if self.value.magnitude_range() == 0..=0 {
    let u8_value = self.value.digit_at(0);
    // ...
}
```

Also, note that this doesn't currently work correctly with negative numbers since we remove the sign earlier now.

Reply:
The values are either -2, -1, 0, 1, 2 in the data. The 0th digit + sign is sufficient.